cdap-cli/src/main/java/co/cask/cdap/cli/CLIConfig.java

```diff
@@ -308,21 +291,33 @@ public void addHostnameChangeListener(ConnectionChangeListener listener) {
    * Listener for hostname changes.
    */
   public interface ConnectionChangeListener {
-    void onConnectionChanged(String currentNamespace, URI newURI);
+    void onConnectionChanged(ClientConfig clientConfig);
   }
 
   /**
    * Connection information to a CDAP instance.
    */
   public static final class ConnectionInfo {
+
     private final String hostname;
     private final int port;
     private final boolean sslEnabled;
+    private final Id.Namespace namespace;
 
-    public ConnectionInfo(String hostname, int port, boolean sslEnabled) {
```

Comment:
Should probably also have a constructor here that does not accept ``Id.Namespace`` and delegates to this constructor using ``Constants.DEFAULT_NAMESPACE_ID``?